### ISSUE TITLE
[BUG] viewportSize() returns null when connecting via CDP

### ISSUE THREAD

**Issue by pohlt** [microsoft/playwright#7114]
**Context:**
- Playwright Version: 1.11.0, 1.12.1 (others not tested)
- Operating System: Windows 10, 64 bit
- Browser: Chromium
- Python 3.7.9, 64 bit

**Code Snippet**

Connect to a running Chromium instance with an open CDP port (`chromium --remote-debugging-port=9222`) and print the viewport size:

```python
  async def main():
        async with async_playwright() as p:
            chromium = await p.chromium.connect_over_cdp("http://localhost:9222")
            context = chromium.contexts[0]
            page = context.pages[0]
            print(page.viewport_size)
            await chromium.close()
```

**Describe the bug**

Instead of the expected dictionary for the `viewport_size`, I get `None`.

**Comment by mxschmitt** [comment]
This is working for now as intended. Null/None means we didn't set it ourselves, you can set it manually, then its also returned from viewport_size.

**Comment by pohlt** [comment]
Thanks for the explanation. This raises a few more questions:

* From what I understand, Playwright's intention is to tell me that I haven't set the viewport by returning a `None`. Well, I already knew that, because... I did not set the viewport. What's the use case?

* And if I set the viewport, I could ask about it a few lines later. Why would I ask for a value I just set?

* When Playwright launches the browser and I don't (explicitly) set the viewport, I get the viewport size:
  ```python
  async def main():
      async with async_playwright() as p:
          chromium = await p.chromium.launch()
          page = await chromium.new_page()
          print(page.viewport_size)
          await chromium.close()
  ```
  Wouldn't it be favorable if both scenarios (internal and external browser launch) behaved as coherently as possible after I'm connected to the browser?

* Other attributes like `context` or `page` are set correctly when I connect to a running browser via CDP. What makes `viewport_size` so special or difficult to set to a more useful and correct value?

* Finally, how can I get the viewport size without setting it? Bypass Playwright and use CDP directly?

**Comment by mxschmitt** [comment]
When Playwright will launch the browser, we always set the viewport to a default value. The user can override it and pass None, then its also None when you call `page.viewport_size`. For the other points, see the first point, normally we set it since we have defaults. I get your reasoning, about it, and would keep it open as a feature request.

What is your use-case for getting the viewport size?

**Comment by pohlt** [comment]
Thanks for reconsidering this topic! 👍🏼 

We develop an app which runs Chromium to display its UI.

In the deployment case, the screen size is given and the browser is started in kiosk mode (which includes fullscreen).

In the test pipeline, the screen size may vary (the developer might run tests locally on her 21:9 screen). Since we do screen compares as a last resort, the viewport must be equal to the deployment viewport size. Kiosk/fullscreen wouldn't work. So we run the regular deployment startup with slightly modified Chromium settings (no kiosk, but open CDP port), connect with Playwright, and run the tests.

But before we can run the tests, we have to make sure that the viewport has the correct size. Since I don't know how to start Chromium with a given viewport size (!= window size), I was trying to get the windows size and the viewport size (here's the root cause for this issue) and modify the window size (using [SetWindowPos](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowpos)) to get the desired viewport size. Fishy, I know, but the best I could come up with.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

We are prioritizing the bugs based on the upvotes, recency and our ability to act. It looks like this issue only has a handful of upvotes, has not been touched recently and/or we lack sufficient feedback to act on it. We are closing issues like this one to keep our bug database maintainable. Please feel free to open a new issue and link this one to it if you think this is a mistake.
